src/rudderstack.ts

```diff
@@ -44,6 +44,7 @@ export class RudderStack {
 
     /** For caching mechanism, Rudderstack  SDK, first page load  */
     handleCachedEvents = () => {
+        const domain = window.location.hostname.includes('deriv.team') ? '.deriv.team' : '.deriv.com'
```

Comment:
```suggestion
        const domain = `.${window.location.host}`
```

Reply:
we can make it generic but once we were asked only to add the deriv.com domain for the security purpose.

Reply:
Noted. then it's fine.

Reply:
@aswathy-deriv  we might need to add more in the future. Can we have an array list of allowed domains so it will be easier for us add later?

Reply:
@prince-deriv So as @shayan-deriv mentioned why don't we make it generic rather than adding the allowed domains?let me know this if not i will add the array and restrict to the allowed domains

Reply:
if let's say we have 3 or more supported domains. Our condition will go longer and longer. 

Here's an idea: 

```
const allowedDomains = ['deriv.com', 'deriv.team', 'anotherdomain.com'];

const domain = allowedDomains.some(d => window.location.hostname.includes(d))
  ? `.${allowedDomains.find(d => window.location.hostname.includes(d))}`
  : `.${allowedDomains[0]}`;
  ```